roles/edpm_network_config/tasks/network_config.yml

```diff
@@ -0,0 +1,149 @@
+    name: edpm_bootstrap
+    tasks_from: packages.yml
+
+- name: Ensure /var/lib/edpm-config directory exists
+  become: true
+  ansible.builtin.file:
+    path: /var/lib/edpm-config
+    state: directory
+    setype: container_file_t
+    selevel: s0
+    mode: 0750
+
+- name: Check for previous run of NetworkConfig
+  become: true
+  ansible.builtin.stat:
+    path: /var/lib/edpm-config/os-net-config.returncode
+  register: os_net_config_returncode_stat
+
+- name: Check result of previous run of NetworkConfig
+  become: true
+  ansible.builtin.slurp:
+    path: /var/lib/edpm-config/os-net-config.returncode
+  when: os_net_config_returncode_stat.stat.exists
+  register: os_net_config_returncode_slurp
+
+- name: Ensure we get cloud-init
+  ansible.builtin.stat:
+    path: /etc/cloud/cloud.cfg.d
+  register: cloud_init_exists
+
+- name: NetworkConfig
+  become: true
+  # The conditions here are when we want to apply the
+  # NetworkConfig. They are:
+  # - If edpm_network_config_update is True
+  # - Or the previous run of NetworkConfig failed.
+  # - Or it has never run
+  # This will match the prior behavior of when a Heat
+  # SoftwareDeployment was used.
+  # It also ensures the script does exist as a sine qua non
+  # condition
+  when:
+    - (edpm_network_config_update) or
+      (os_net_config_returncode_stat.stat.exists and
+      ((os_net_config_returncode_slurp.content | b64decode | int) != 0)) or
+       (not os_net_config_returncode_stat.stat.exists)
+  block:
+    - name: Create /etc/os-net-config directory
+      become: true
+      ansible.builtin.file:
+        path: /etc/os-net-config
+        state: directory
+        recurse: true
+
+    - name: Create os-net-config mappings from lookup data
+      edpm_os_net_config_mappings:
+        net_config_data_lookup:
+          "{{ edpm_network_config_os_net_config_mappings }}"
+      when: not ansible_check_mode|bool
+      register: os_net_config_mappings_result
+
+    - name: Write os-net-config mappings file /etc/os-net-config/mapping.yaml
+      ansible.builtin.copy:
+        content: "{{ os_net_config_mappings_result.mapping | to_nice_yaml }}"
+        dest: /etc/os-net-config/mapping.yaml
+        mode: 0644
+      when: os_net_config_mappings_result.changed|bool
+
+    - name: Manage NetworkConfig with edpm_os_net_config module
+      block:
+        - name: Remove /var/lib/edpm-config/scripts directory
+          ansible.builtin.file:
+            path: /var/lib/edpm-config/scripts
+            state: absent
+
+        - name: Run NetworkConfig with edpm_os_net_config
+          ansible.builtin.include_tasks: os_net_config.yml
+
+    - name: Write rc of NetworkConfig script
+      ansible.builtin.copy:
+        content: "{{ network_config_result.rc }}"
+        dest: /var/lib/edpm-config/os-net-config.returncode
+        mode: 0644
+      when:
+        - network_config_result.rc is defined
+
+    # LP Bug: https://bugs.launchpad.net/edpm/+bug/1958332
```

Comment:
This bug doesn't exist. Because it used to say tripleo. I suspect this is outcome of sed application. Maybe we should link back to the original [1]. Or we can replace this with a comment explaining why is this being done.

[1]https://bugs.launchpad.net/tripleo/+bug/1958332